### PR TITLE
Feat:  groq API integration

### PR DESCRIPTION
**Pull request by PranotiPD**
Closes:  #205 

- **Title** : groq API integration
- **Name:**  Pranoti Durugkar
- **Idenitfy yourself:**  Gssoc

### Describe the add-ons or changes you've made 📃

I have added a page under the resources section where user can put their question and search for answers and user will be answered as if it is being answered by Chanakya.

### Type of change ☑️

What sort of change have you made:

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Code style update (formatting, local variables)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

### Checklist: ☑️

- [x] My code follows the [Contributing Guidelines](https://github.com/Avdhesh-Varshney/Chanakya-Niti/blob/main/README.md) & [Code of Conduct](https://github.com/Avdhesh-Varshney/Chanakya-Niti/blob/main/CODE_OF_CONDUCT.md) of this project.
- [x] This PR does not contain plagiarized content.
- [x] I have performed a self-review of my own code.
- [x] I have commented my code, particularly wherever it was hard to understand.
- [x] My changes generate no new warnings.

### CODE REVIEW

File: src/components/shared/Footer.jsx [hunk 1]
```diff
@@ -37,6 +37,7 @@ const Footer = () => {
           <div className="col-md-3 my-auto">
             <h3 className="text-uppercase fw-bold">Resources</h3>
             <ul className="list-unstyled">
+            <li><NavLink to="/resources/audio" className="text-decoration-none text-dark"><FaMusic className="me-2" /> ChanakyaGpt</NavLink></li>
```
Comment (Avdhesh-Varshney) [inline src/components/shared/Footer.jsx:40]: image

fix routing

File: src/components/shared/Navbar.jsx [hunk 2]
```diff
@@ -47,6 +47,7 @@ const Navbar = () => {
                 Resources
               </Link>
               <ul className="dropdown-menu" aria-labelledby="navbarDropdown">
+                <li><Link className="dropdown-item" to="/resources/chanakyaGpt">ChanakyaGpt</Link></li>
```
Comment (Avdhesh-Varshney) [inline src/components/shared/Navbar.jsx:50]: image

fix routing

**Review by Avdhesh-Varshney — CHANGES_REQUESTED** [review]
@PranotiPD LGTM.

**Comment by Avdhesh-Varshney** [comment]
Always link your issue not the pr. Update it.

**Comment by PranotiPD** [comment]
@Avdhesh-Varshney made routing changes

**Review by Avdhesh-Varshney — APPROVED** [review]

**Pull request merged**